shapely/errors.py

```diff
@@ -41,3 +41,10 @@ class ShapelyDeprecationWarning(FutureWarning):
 
 class EmptyPartError(ShapelyError):
     """An error signifying an empty part was encountered when creating a multi-part."""
+
+
+class GeometryTypeError(ShapelyError):
+    """
+    An error raised when the type of the geometry in question is
+    unrecognized or inappropriate
+    """
```

Comment:
```suggestion
class GeometryTypeError(TypeError, ValueError):
    """
    An error raised when the type of the geometry in question is
    unrecognized or inappropriate.
    """
    def __init__(self, msg):
        warnings.warn("", ShapelyDeprecationWarning, stacklevel=2)
        super()
```

Reply:
Ah, good point! Wasn't thinking about backwards compatibility. Makes sense to me 👍